src/Evolution/Systems/GeneralizedHarmonic/Constraints.hpp

```diff
@@ -180,4 +180,74 @@ void four_index_constraint(
     gsl::not_null<tnsr::iaa<DataType, SpatialDim, Frame>*> constraint,
     const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept;
 // @}
+
+// @{
+/*!
+ * \brief Computes the generalized-harmonic (unnormalized) constraint energy.
+ *
+ * \details Computes the generalized-harmonic unnormalized constraint energy
+ * [Eq. (53) of \cite Lindblom:2005qh with
+ * \f$m^{ab}=\delta^{ab}\f$ and with each term in the sum scaled by an
+ * arbitrary coefficient],
+ * \f{eqnarray}{
+ * E & = & K_1 C_a C_a + K_2\left(F_a F_a
+     + C_{ia} C_{ja} g^{ij}\right)\nonumber\\
+     & + & K_3 C_{iab} C_{jab} g^{ij} + K_4 C_{ikab} C_{jlab}g^{ij} g^{kl}.
+ * \f}
+ * Here \f$C_a\f$ is the gauge constraint, \f$F_a\f$ is the f constraint,
+ * \f$C_{ia}\f$ is the two-index constraint, \f$C_{iab}\f$ is the
+ * three-index constraint, \f$C_{ikab}\f$ is the four-index constraint,
+ * \f$g^{ij}\f$ is the inverse spatial metric, and
+ * \f$K_1\f$, \f$K_2\f$, \f$K_3\f$, and \f$K_4\f$ are constant multipliers
+ * for each term that each default to a value of 1.0. Note that in this
+ * equation, spacetime indices \f$a,b\f$ are raised and lowered with
+ * the Kronecker delta.
+ *
+ * Also note that the argument `four_index_constraint` is a rank-3 tensor.
+ * This is because `GeneralizedHarmonic::four_index_constraint()` takes
+ * advantage of the antisymmetry of the four-index constraint's first two
+ * indices to only compute and return the independent
+ * components of \f$C_{ijab}\f$, which can be written as
+ * \f{eqnarray}{
+ * D_{iab} \equiv \frac{1}{2} \epsilon_{i}{}^{jk} C_{jkab},
+ * \f} where \f$\epsilon_{ijk}\f$ is the flat-space Levi-Civita tensor,
+ * whose inidces are raised and lowered with the Kronecker delta.
+ * The note https://www.overleaf.com/read/jfvzkkywqcqx derives the
```

Comment:
I'm not sure I like linking to external documents like this. You can convert the latex code to doxygen using `tools/latex2dox.py`.

Reply:
It's a 4-page note with 24 equations. Are you saying you want that much help text here?

Reply:
Yep, that's exactly what I'm saying. This was done for Kerr-Schild as well: https://spectre-code.org/classgr_1_1solutions_1_1kerrschild